src/ReusableComponents/appBar.js

```diff
@@ -41,37 +42,73 @@ const styles = {
-                        aria-label="Open drawer"
-                        onClick={this.props.handleDrawerOpen}
-                        className={classNames(classes.menuButton, open)}
-                    >
-                        <MenuIcon
-                            className={classes.menuButtonIcon}
-                        />
-                    </IconButton>
-                    <div className={classes.grow}>
+        const { classes,open,history } = this.props;
+        const { isTitleEditable } = this.state;
+        return (
+            <div className={classes.root}>
+                <AppBar position="static">
+                    <Toolbar>
                         <IconButton
-                            aria-label="Go back one page"
-                            onClick={history.goBack}
-                            className={classes.backButton}
+                            aria-label="Open drawer"
+                            onClick={this.props.handleDrawerOpen}
+                            className={classNames(classes.menuButton, open)}
                         >
-                            <BackIcon className={classes.backButtonIcon} />
+                            <MenuIcon
+                                className={classes.menuButtonIcon}
+                            />
                         </IconButton>
-                        <Typography variant="h6" color="inherit">
-                            {this.props.title}
-                        </Typography>
-                    </div>
-                </Toolbar>
-            </AppBar>
-        </div>
-    );
+                        <div className={classes.grow}>
+                            <IconButton
+                                aria-label="Go back one page"
+                                onClick={history.goBack}
+                                className={classes.backButton}
+                            >
+                                <BackIcon className={classes.backButtonIcon} />
+                            </IconButton>
+                            {isTitleEditable && this.props.editable ? (
+                                <form onSubmit={this.handleTitleChange} >
+                                    <InputBase
+                                        autoFocus
+                                        type="input"
+                                        placeholder={this.props.title}
+                                        value={this.state.newTitle}
+                                        onChange={(e) => this.setState({ newTitle: e.target.value })}
+                                        onBlur={this.handleTitleChange}
+                                    />
+                                </form>
+                            ) : (
+                                <Typography variant="h6" color="inherit" onClick={this.toggleEditableTitle}>
+                                    {this.props.title}
+                                </Typography>
```

Comment:
I think it would be a good idea to add some visual indicators showing that the title is editable. Without them, the user might not think to click on the title.

2 ideas:
- Style this `<Typography>` component to have [`cursor: pointer`](https://developer.mozilla.org/en-US/docs/Web/CSS/cursor). This will change the mouse icon on hover to show that the title is interactive.
- Wrap this `<Typography>` component in a [`<Tooltip>`](https://v3.material-ui.com/demos/tooltips/#tooltips) that shows a message like "Click to rename" on hover. An example of `<Tooltip>` in this repo can be found [here](https://github.com/averi-studios/world-scribe-2-desktop/blob/a8a90d48d700496471b4a4d9edc9a4780fc2c599/src/ReusableComponents/sidePanel.js#L140).

If you could implement both, that would be great. Also, both of these will need a conditional check so that the they're only visible when the title is editable.

Reply:
Added both pointer and tooltip